fisco-bcos/benchmark/table_benchmark.cpp

```diff
@@ -0,0 +1,252 @@
+ * along with FISCO-BCOS.  If not, see <http://www.gnu.org/licenses/>
+ * (c) 2016-2018 fisco-dev contributors.
+ *
+ * @file kvtable_benchmark.cpp
+ * @author: xingqiangbai
+ * @date 2020-03-18
+ */
+
+#include "libinitializer/Initializer.h"
+#include "libledger/DBInitializer.h"
+#include "libprecompiled/KVTableFactoryPrecompiled.h"
+#include "libprecompiled/TableFactoryPrecompiled.h"
+#include "libstorage/BasicRocksDB.h"
+#include "libstorage/MemoryTableFactory2.h"
+#include "libstorage/MemoryTableFactoryFactory2.h"
+#include "libstorage/RocksDBStorage.h"
+#include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string/classification.hpp>
+#include <boost/algorithm/string/split.hpp>
+#include <boost/program_options.hpp>
+#include <cstdlib>
+#include <functional>
+
+using namespace std;
+using namespace dev;
+using namespace boost;
+using namespace dev::db;
+using namespace dev::ledger;
+using namespace dev::storage;
+using namespace dev::blockverifier;
+using namespace dev::initializer;
+using namespace dev::precompiled;
+
+namespace po = boost::program_options;
+
+po::options_description main_options("Main for Table benchmark");
+
+po::variables_map initCommandLine(int argc, const char* argv[])
+{
+    main_options.add_options()("help,h", "help of Table benchmark")("accounts,a",
+        po::value<int>()->default_value(50),
+        "accounts with write permission, if 0 every account has write permission")("path,p",
+        po::value<string>()->default_value("benchmark/table/"),
+        "[RocksDB path]")("cache,c", po::value<int>()->default_value(0),
+        "memory size(MB) of CachedStorage, if 0 then no CachedStorage")(
+        "keys,k", po::value<int>()->default_value(10000), "the number of different keys")("value,v",
+        po::value<int>()->default_value(128),
+        "the length of value")("random,r", "every time use a new rocksdb");
+    po::variables_map vm;
+    try
+    {
+        po::store(po::parse_command_line(argc, argv, main_options), vm);
+        po::notify(vm);
+    }
+    catch (...)
+    {
+        std::cout << "invalid input" << std::endl;
+        exit(0);
+    }
+    if (vm.count("help") || vm.count("h"))
+    {
+        std::cout << main_options << std::endl;
+        exit(0);
+    }
+    return vm;
+}
+
+int main(int argc, const char* argv[])
+{
+    boost::property_tree::ptree pt;
+    auto logInitializer = std::make_shared<LogInitializer>();
+    logInitializer->initLog(pt);
+    auto params = initCommandLine(argc, argv);
+    uint32_t authorityAccounts = params["accounts"].as<int>();
+    auto storagePath = params["path"].as<string>();
+    if (params.count("random"))
+    {
+        storagePath += to_string(utcTime());
+    }
+    auto cacheSize = params["cache"].as<int>();
+    bool useCachedStorage = cacheSize == 0 ? false : true;
+    auto keys = params["keys"].as<int>();
+    auto valueLength = params["value"].as<int>();
+    int16_t groupID = 1;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The scope of the variable 'groupID' can be reduced.](https://app.codacy.com/gh/FISCO-BCOS/FISCO-BCOS/pullRequest?prid=5208383)